packages/docusaurus-theme-classic/src/theme/hooks/useLocationHash.ts

```diff
@@ -5,9 +5,11 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import {useState, useEffect} from 'react';
+import {useState, useEffect, Dispatch, SetStateAction} from 'react';
 
-function useLocationHash(initialHash) {
+function useLocationHash(
+  initialHash: string,
+): [string, Dispatch<SetStateAction<string>>] {
```

Comment:
Subjective trick: if you do `return [hash, setHash] as const`, you don't need to add explicitly the hooks return type (but having it is fine too anyway)

Reply:
It is there to make linter happy (explicit return type).